hs_core/hydroshare/utils.py

```diff
@@ -432,17 +596,64 @@ def resource_file_add_process(resource, files, user, extract_metadata=False, **k
     return resource_file_objects
 
 
-def add_file_to_resource(resource, f):
+def add_file_to_resource(resource, f, fed_res_file_name_or_path='', fed_copy=None):
```

Comment:
I agree less parameter is better. I am OK with the current implementation. My concern was related to ```fed_copy = False``` being interpreted as a move operation. I would prefer to name the parameter as  ```fed_copy_or_move``` with the following possible values:

```fed_copy_or_move = 'copy'```:  copy the file
```fed_copy_or_move = 'move'```: move the file
any other value for ```fed_copy_or_move```  No file action needed.